test/test_madx.py

```diff
@@ -718,6 +722,18 @@ def test_multiline_input(self):
         ''')
         self.assertEqual(var.x, 3)
 
+    def test_field_errors(self):
+        mad = self.mad
+        mad.beam()
+        mad.use(sequence='s1')
+        mad.select(flag='error', range='qp')
+        mad.efcomp(dkn=[1e-6, 2e-6, 3e-6], dks=[-1e-6, -2e-6, -3e-6])
+        mad.ealign(dx=1e-3,dy=-4e-3)
+        assert mad.sequence['s1'].expanded_elements['qp'].field_errors[0]==1e-6
```

Comment:
best check for `np.allclose()`

Reply:
> > What do you think returning `namedtuples` or dict instead of bare arrays like:
> 
> Sounds good to me. You may also consider using `AttrDict` instead (move it to cpymad/types.rst first) which has the advantage of being to unpack directly (i.e. `**field_errors`), as well as case-insensitive access. Choose whichever you prefer.

AttrDict should be implemented in madx.py, right?

Do you advantage/disadvantages in using cython classes in libmadx?

```
cdef class FieldError:
    cdef public object dkn
    cdef public object dks

cdef class AlignError:
    cdef public double dx
    cdef public double dy
    ...

cdef class PhaseError:
    cdef public object dpn
    cdef public object dps

```

Reply:
Might be fine too, but I think you have to do additional work to expose the attribute names dkn/dks to the user. AttrDict is currently in madx.py but if it were returned from libmadx routines, it should be moved to types.py so both can import it.

Reply:
In fact, no don't use cdef classes right now. They won't be transferred by the IPC library. Maybe in the future it will be an option.